types/openid/index.d.ts

```diff
@@ -9,19 +10,25 @@ export interface OpenIdError {
 }
 
 export class RelyingParty {
-  constructor(
-    returnUrl: string,
-    realm: string | null,
-    stateless: boolean,
-    strict: boolean,
-    extensions: readonly any[]);
+           constructor(
+               returnUrl: string,
+               realm: string | null,
+               stateless: boolean,
+               strict: boolean,
+               extensions: readonly any[]
+           );
 
-  authenticate(
-    identifier: string,
-    immediate: boolean,
-    callback: (err: OpenIdError | null, authUrl: string | null) => void): void;
+           authenticate(
+               identifier: string,
+               immediate: boolean,
+               callback: (err: OpenIdError | null, authUrl: string | null) => void
+           ): void;
 
-  verifyAssertion(
-    requestOrUrl: object | string,
-    callback: (err: OpenIdError | null) => void): void;
-}
+           verifyAssertion(
+               requestOrUrl: object | string,
+               callback: (
+                   err: OpenIdError | null,
+                   result?: { authenticated: boolean; claimedIdentifier?: string }
```

Comment:
As discussed [here](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#optional-parameters-in-callbacks), we recommend not using optional parameters in callbacks.

Reply:
If there is an `err` then `result` won't be set. What do you suggest?

Reply:
That's fine, then. Thanks!